cats-effect/src/test/scala/org/specs2/cats/effect/CatsResourceSpec.scala

```diff
@@ -0,0 +1,68 @@
+package org.specs2.cats.effect
+
+import cats.effect.IO
+import cats.effect.Ref
+import cats.effect.Resource
+import org.specs2.Specification
+import org.specs2.execute.Result
+
+trait LocalCatsResourceSpec extends Specification, IOExecution, IOMatchers, CatsResource[(Int, Int)]:
+  import LocalCatsResourceSpec.*
+
+  val localCounter = Ref.unsafe[IO, Int](0)
+  val released = Ref.unsafe[IO, Boolean](false)
+
+  def resource: Resource[IO, (Int, Int)] = Resource
+    .eval(localCounter.getAndUpdate(_ + 1).product(globalCounter.getAndUpdate(_ + 1)))
+    .onFinalize(released.set(true))
+
+  def is = s2"""
+    The resource is re-acquired per-spec $reacquired
+    The resource is shared in the spec $shared
+    The resource is shared in the spec $shared
```

Comment:
You added an other example on purpose?

Reply:
Yes, the reason I added it twice is to check that the resource has not been re-acquired (since it increases a counter every time that it is acquired). You are right, I think checking this is equivalent to checking that they can both modify the resource (and see each other's modifications). I'm just not used to thinking about global mutable state 😀 Thanks, let me fix this up!